src/main/database.cpp

```diff
@@ -21,6 +22,20 @@ using namespace kuzu::transaction;
 namespace kuzu {
 namespace main {
 
+void getFileFlags(SystemConfig systemConfig, int& flags, FileLockType& lock, bool create_new) {
+    if (systemConfig.access_mode == AccessMode::READ_ONLY) {
+        assert(!create_new);
+        flags = O_RDONLY;
+        lock = FileLockType::READ_LOCK;
+    } else {
+        flags = O_RDWR;
+        lock = FileLockType::WRITE_LOCK;
+        if (create_new) {
+            flags = O_WRONLY | O_CREAT;
+        }
+    }
```

Comment:
If it's createNew, do we need to set flags to `O_WRONLY`?, will `O_RDWR` be fine?
If so, I think this piece of code can be re-organized as following:
```cpp
    flags = systemConfig.access_mode == AccessMode::READ_ONLY ? O_RDONLY : O_RDWR;
    if (create_new) {
        assert(flags == O_RDWR);
        flags |= O_CREAT;
    }
    lock = systemConfig.access_mode == AccessMode::READ_ONLY ? FileLockType::READ_LOCK : FileLockType::WRITE_LOCK;
```